postgres/tests/test_statements.py

```diff
@@ -1286,16 +1290,31 @@ def test_pg_settings_caching(aggregator, integration_check, dbm_instance):
     ), "key should not have been blown away. If it was then pg_settings was not cached correctly"
 
 
+def _check_until_time(check, dbm_instance, sleep_time, check_interval):
+    start_time = time.time()
+    elapsed = 0
+    # Keep calling check to avoid triggering check inactivity
+    while elapsed < sleep_time:
+        check.check(dbm_instance)
+        time.sleep(check_interval)
+        elapsed = time.time() - start_time
+
+
 def test_statement_samples_main_collection_rate_limit(aggregator, integration_check, dbm_instance):
     # test the main collection loop rate limit
     collection_interval = 0.1
+    # Don't need query metrics or activity for this one
+    dbm_instance['query_metrics']['enabled'] = False
```

Comment:
this is a great update. Did you look through other tests for opportunities to do this elsewhere? if not i'll make a jira ticket for us to make this improvement across the rest of our tests where possible!

Reply:
I've done a pass on the whole file while I was on it and disabled unnecessary jobs.
Locally, it squeezed some additional seconds from ~20s to ~16.5s and most of the tests are now under the seconds. 

```
1.11s call     tests/test_statements.py::test_activity_collection_rate_limit
1.04s call     tests/test_statements.py::test_statement_samples_main_collection_rate_limit
0.38s call     tests/test_statements.py::test_statement_metrics[False-datadog.pg_stat_statements()-False-ignore_databases1]
0.36s call     tests/test_statements.py::test_statement_metrics[True-pg_stat_statements-False-ignore_databases1]
0.36s call     tests/test_statements.py::test_async_job_inactive_stop
```

On the CI, it seems like we're still around the same ~16m of tests.